tests/taskcluster/bin/raptor_graph

```diff
@@ -29,48 +37,115 @@ function params() {
   }, {});
 }
 
-// Map of task ids to task names.
+// Map of task ids to task names
 var taskIdMap = {};
 
+/**
+baseTask: run test on 'base' build
+patchTask: run test on 'patch' build
+resultTask: compare results, check for regression
+
+'base build' will always be the provided BASE_REV
+
+'patch build' will change accordingly:
+if testing a PULL_REQUEST 'patch build' will be the pr/
+provided HEAD_REV; else it will be BASE_REV -1 commit
+*/
+
 var config = {
-  task1: {
+  baseTask1: {
     definition: 'raptor_launch',
     params: {
-      provisionerId: 'aws-provisioner-v1',
-      workerType: 'gaia',
-      raptorRuns: 15,
-      raptorApps: 'settings',
+      raptorSuiteName: 'Settings Launch Test (on base)',
       gaiaRev: process.env.GITHUB_BASE_REV,
-      raptorTimeout: 600000,
-      raptorBootSleep: 180,
+      raptorRuns: 30,
+      raptorApps: 'settings',
       treeherderSym: 'se1'
     }
   },
-  task2: {
+  patchTask1: {
     definition: 'raptor_launch',
     params: {
-      provisionerId: 'aws-provisioner-v1',
-      workerType: 'gaia',
-      raptorRuns: 15,
-      raptorApps: 'settings',
+      raptorSuiteName: 'Settings Launch Test (on base + pr)',
       gaiaRev: process.env.GITHUB_HEAD_REV,
-      raptorTimeout: 600000,
-      raptorBootSleep: 180,
+      raptorRuns: 30,
+      raptorApps: 'settings',
       treeherderSym: 'se2'
     }
   },
-  task3: {
+  resultTask: {
     definition: 'raptor_result',
+    requires: ['baseTask1', 'patchTask1'],
     params: {
-      completedBranchTask: '',
-      completedPatchTask: '',
-      provisionerId: 'aws-provisioner-v1',
-      workerType: 'gaia',
+      raptorSuiteName: 'Settings Results',
       treeherderSym: 'se'
     }
   }
 };
 
+/**
+ * Execute the built command in a child process
+ * @returns {Promise}
+ */
+var do_command = function(cmd) {
+  return new Promise(function(resolve, reject) {
+    exec(cmd, function(err, stdout, stderr) {
+      if (err) {
+        return reject(err, stderr);
+      }
+      resolve(stdout.trim());
+    });
+  });
+};
+
+/**
+ * Resolve when latest emulator build url has been retrieved
+ * @returns {Promise}
+ */
+var getEmulator = function() {
+  return new Promise(function(resolve, reject) {
+
+    // Get taskId for the last successful emulator-ics build task
+    var taskId = 99;
+    var ns = "buildbot.branches.mozilla-central.emulator-ics";
+
+    // Find task in namespace
+    var index = new taskcluster.Index(TC_CLIENT_OPTS);
+    index.findTask(ns)
+      .then(function(foundTask) {
+        // Url for build is
+        // https://queue.taskcluster.net/v1/task/{taskId}/artifacts/{artifact}
+        emulatorBuildURL = url.format({
```

Comment:
Right, actually I was trying that route first but couldn't get it to work, probably because I wasn't using the proxy feature. I'll leave it is as-is, this is how mozilla-download does it too. If down the road artifact url structure changes then I could revisit this, thanks